app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\User;
+use App\Http\Requests\UpdateUserRequest;
+use App\Repositories\RepositoryInterface\UserRepositoryInterface;
+
+class UserController extends Controller
+{
+    protected $user;
+
+    public function __construct(UserRepositoryInterface $user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $getUsers = $this->user->getAll();
+
+        return view('user.users', compact('getUsers'));
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $getUser = $this->user->find($id);
+        if(!$getUser) return redirect()->route('admin.users.index')->with('message', ['msg' => trans('admin.user_message.not_found'), 'status' => 'danger']);
+        
+        return view('user.edit-user', compact('getUser'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(UpdateUserRequest $request, User $user)
+    {
+
+        $getUser = $this->user->find($user->user_id);
+        if(!$getUser) return redirect()->route('admin.users.index')->with('message', ['msg' => trans('admin.user_message.not_found'), 'status' => 'danger']);
+        $this->user->update($user->user_id, $request->all());
+
+        return redirect()->route('admin.users.index')->with('message', ['msg' => trans('admin.user_message.edit_success'), 'status' => 'success']);
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        $getUser = $this->user->find($id);
+        if(!$getUser) return redirect()->route('admin.users.index')->with('message', ['msg' => trans('admin.user_message.not_found'), 'status' => 'danger']);
+        $this->user->delete($id);
```

Comment:
Lấy cái đã find ở trên mà delete luôn e. Sao gọi từ $this->user nữa?
K check xóa success hay fail à e?

Reply:
> Lấy cái đã find ở trên mà delete luôn e. Sao gọi từ $this->user nữa?
Dạ em sửa rồi

> K check xóa success hay fail à e?
Có mà anh, nếu không find ra thì quay lại trang users.index rồi hiện thông báo lỗi

Reply:
Xem lại Logic e nhé
1. Thì gọi $this->user->delete($id) từ Repository luôn, không có find ở đây nữa vì trong kia method delete có find rồi.
Bên này chỉ check status trả về xóa thành công hay thất bại thôi
2.  Là dùng như em ở trên thì cái hàm delete ở Repository đâu có tác dụng gì?
Sử theo cái 1

Reply:
dạ ok anh